### ISSUE TITLE
Deployment ignores errors from the Kudu command API

### ISSUE THREAD

**Issue by vjrantal** [serverless/serverless-azure-functions#71]
If there are errors returned from the Kudu command API during deployment, they are silently ignored.

To illustrate what I mean, I applied the following diff to the code:

```
diff --git a/provider/azureProvider.js b/provider/azureProvider.js
index 4c3c4d4..64b67cc 100644
--- a/provider/azureProvider.js
+++ b/provider/azureProvider.js
@@ -542,6 +542,8 @@ return new BbPromise((resolve, reject) => {
     };
 return new BbPromise((resolve, reject) => {
       request(options, (err, res, body) => {
+        console.log('Received status: ' + res.statusCode);
+        console.log(body);
         if (err) {
           reject(err);
         }
```

After that, a certain deploy command printed output like this:

```
Serverless: Syncing Triggers....Response statuscode: 200
Serverless: Running Kudu command del package.json...
Received status: 200
{ Output: '',
  Error: 'Could Not Find D:\\home\\site\\wwwroot\\package.json\r\n',
  ExitCode: 0 }
Serverless: Uploading pacakge.json ...
Serverless: Running Kudu command npm install --production...
Received status: 500
<html><head><title>500 - The request timed out.</title></head><body>  <font color ="#aa0000">         <h2>500 - The request timed out.</h2></font>  The web server failed to respond within the specified time.</body></html>
Serverless: Successfully created Function App
```

In above, observe how in the first call, the status is 200, but error is seen from the content of the body and in the second case, the status is 500. In both cases, the errors are ignored and eventually `Successfully created Function App` is printed even though there were errors.

Fix proposal is to look at both status code and body content in all Kudu commands and reject the related Promise in case errors are identified.

ps. On purpose, I am not providing reproduce steps since I intended this one to be only about proper error handling in case any type of errors occur.

**Comment by fiveisprime** [comment]
Thank you so much for the detailed report! 🎉 

I'll get some additional error handling added and deployed this week (hopefully today).

**Comment by fiveisprime** [comment]
@vjrantal Can you deploy your test/broken function with my fork?

```sh
$ npm install fiveisprime/serverless-azure-functions
```

See if it errors out correctly. I made some changes to that particular piece, but I'm still working through all of the other places where this could come up.

**Comment by lostintangent** [comment]
@fiveisprime Did #72 end up resolving this issue?

**Comment by lostintangent** [comment]
Closing this as resolved by #72. Please reopen if you experience any further issues. Thanks!
